app/controllers/users.rb

```diff
@@ -0,0 +1,42 @@
+get '/users/new' do
+  @user = User.new
+  erb :"users/new"
+end
+
+post '/users' do
+  @user = User.new(params[:user])
+  if @user.save
+    session[:user_id] = @user.id
+    redirect "/users/login"
+  else
+    erb :"users/new"
+  end
+end
+
+get '/login' do
+  erb :"/users/login"
+end
+
+post '/login' do
+  @user = User.authenticate(params[:username], params[:password])
+  if @user
+    session[:user_id] = @user.id
+    redirect "/users/#{@user.id}"
+  else
+    @error = "Invalid login, please try again!"
+    erb :'users/login'
+  end
+end
+
+get '/users/logout' do
+  session.clear
+  redirect "/"
+end
+
+#LAST METHOD
+get '/users/:id' do
+  @user = User.find_by(id: params[:id])
+  @suggestions = Suggestion.find_by(user_id: params[:user_id])
+  redirect '/' unless current_user.id == session[:user_id]
```

Comment:
Heads up: `current_user.id == session[:user_id]` will always be `true`, since `current_user` is referencing `session[:user_id]`. What I think you meant to write is: 
```ruby
redirect '/' unless @user.id == current_user.id
```